sympy/polys/polytools.py

```diff
@@ -4427,6 +4427,10 @@ def degree(f, gen=0):
         if gen not in p.gens:
             return S.Zero
```

Comment:
your added code should be an elif clause:

```
if not sympify(gen, strict=True).is_Number:
    ...
elif not f.is_Poly and len(f.free_symbols) > 1:
    raise TypeError(filldedent('''
            A symbolic generator of interest is required when passing
            a multivariate expression like foo = %s, e.g. degree(foo, gen=%s)
            instead of degree(foo, gen=%s).
        ''' % (f, ordered(f.free_symbols).next(), gen)))
```

Note that the filldedent text comes on the next line so the error appears more neatly during interactive sessions. Also, an example specific to the user's passed multivariate expression is given instead of a generic expression.

e.g.
```
>>> f=x+y**2
>>> gen=1
>>> raise TypeError(filldedent('''
...             A symbolic generator of interest is required when passing
...             a multivariate expression like foo = %s, e.g. degree(foo, gen=%s
)
...             instead of degree(foo, gen=%s).
...         ''' % (f, ordered(f.free_symbols).next(), gen)))
Traceback (most recent call last):
  File "<stdin>", line 5, in <module>
TypeError:
A symbolic generator of interest is required when passing a
multivariate expression like foo = x + y**2, e.g. degree(foo, gen=x)
instead of degree(foo, gen=1).
```
If you use this, be sure that `ordered` is imported at the top of the file.